storage/gcloud/aio/storage/blob.py

```diff
@@ -120,12 +132,30 @@ async def get_signed_url(  # pylint: disable=too-many-locals
 
         str_to_sign = '\n'.join(['GOOG4-RSA-SHA256', request_timestamp,
                                  credential_scope, canonical_req_hash])
-        signed_resp = await iam_client.sign_blob(
-            str_to_sign, service_account_email=service_account_email,
-            session=session)
 
-        signature = binascii.hexlify(
-            decode(signed_resp['signedBlob'])).decode()
+        marker_id, key_bytes = pem.readPemBlocksFromFile(
+            io.StringIO(key), PKCS1_MARKER, PKCS8_MARKER)
+        # Given `pem.readPemBlocksFromFile(key, *args)`, returns `marker_id`
+        # as the index of the args which matched or -1 for an invalid/unmatched
+        # key.
+        if marker_id == -1:
+            raise ValueError('private key is invalid or unsupported')
+
+        if marker_id == 1:
+            # pkcs8 -> pkcs1
+            key_info, remaining = decoder.decode(key_bytes,
+                                                 asn1Spec=PKCS8_SPEC)
+            if remaining != b'':
+                raise ValueError('could not read PKCS8 key: found extra bytes',
+                                 remaining)
+
+            private_key_info = key_info.getComponentByName('privateKey')
+            key_bytes = private_key_info.asOctets()
+
+        key = rsa.key.PrivateKey.load_pkcs1(key_bytes, format='DER')
```

Comment:
nit: The `key` variable was used above on line 93. Since this is related to cryptography and the same variable stores two different types of data, a different variable name may help make the code more clear.